src/app/common/theme-provider.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type ComputedTheme = 'Dark' | 'Light';
+export type UserSelectedTheme = ComputedTheme | 'System';
```

Comment:
Small suggestion but one I reckon is important. I'd recommend we decouple the two things that these string types represent:

1) A unique theme identifier string
2) The string of the label presented to the user

Here, we use the same for both, which works. But the coupling can cause issues down the line. Consider we persist the current theme in storage, then decide to rename the theme? Now there store-level concerns for a simple copy edit.

Elsewhere we use lower (snake) case for [identifier](https://github.com/hirosystems/stacks-wallet-web/blob/eb836d5b9bf3de7d80f860d32cfe972ca7dd7275/src/shared/signature/types.ts#L2-L3) [strings](https://github.com/hirosystems/stacks-wallet-web/blob/eb836d5b9bf3de7d80f860d32cfe972ca7dd7275/src/app/common/use-wallet-type.ts#L4), (as these are just single words, I'd lower case them):


```suggestion
type ComputedTheme = 'dark' | 'light';
export type UserSelectedTheme = ComputedTheme | 'system';
```

To get the label, either use some parsing function, or create a map between the id and label:

```ts
type ThemeLabelMap = Record<UserSelectedTheme, string>;
const labelMap: ThemeLabelMap = {
  dark: 'Dark',
  // etc...
};
function getThemeLabel(theme: UserSelectedTheme) {
  return labelMap[theme];
}
```